### PR TITLE
install docs: developer info

### PR DESCRIPTION
**Pull request by DropD**
* add a FAQ item about requirements.txt
* add developer install tipps subsection with info on how to work with aiida using development branches

### CODE REVIEW

File: docs/source/installation.rst [hunk 1]
```diff
@@ -62,6 +62,16 @@ There are many projects that aim to make managing many virtual environments easi
 
 Tab-completion is only supported for bash.
 
+Where did requirements.txt go?
+------------------------------
+
+The recommended way of installing dependencies is now to use `python-pip_`.
+The information that used to be in ``requirements.txt`` is now in ``setup_requirements.py``. The script ``aiida_core/utils/create_requirements.py`` generates a ``requirements.txt`` file from it.
+
+If you are wondering how to update dependencies after pulling a newer instable verion from the repository, have a look at `install.other.devel_`
```
Comment (nmounet) [inline docs/source/installation.rst]: instable -> stable (I guess)

File: docs/source/installation.rst [hunk 2]
```diff
@@ -267,6 +278,29 @@ Next steps:
...
+
+   $ cd aiida_core
+   $ git checkout develop
+   $ pip install -e .[REST]
+
+The same applies for feature branches.
+
+The following points may be helpful:
+
+* When switching to using virtual environments make sure to outcomment your path settings for older aiida installs.
```
Comment (nmounet) [inline docs/source/installation.rst]: aiida -> AiiDA

**Comment by DropD** [comment]
i changed the pull request to be into documentation, rebased and made the suggested changes.

**Comment by DropD** [comment]
The changes have moved into installation/index.rst into the updating from previous versions section

**Pull request merged**